front_end/src/components/onboarding/OnboardingCheck.tsx

```diff
@@ -1,39 +1,45 @@
 "use client";
 
-import { useSearchParams, useRouter } from "next/navigation";
-import { useEffect, useCallback } from "react";
+import { usePathname } from "next/navigation";
+import { useEffect } from "react";
 
 import { useAuth } from "@/contexts/auth_context";
 import { useModal } from "@/contexts/modal_context";
+import { useNavigation } from "@/contexts/navigation_context";
 
 const OnboardingCheck: React.FC = () => {
-  const searchParams = useSearchParams();
-  const router = useRouter();
   const { setCurrentModal } = useModal();
   const { user } = useAuth();
-
-  const handleOnboarding = useCallback(() => {
-    const startOnboarding = searchParams.get("start_onboarding");
-    if (startOnboarding === "true") {
-      // Remove the query parameter
-      const newUrl = new URL(window.location.href);
-      newUrl.searchParams.delete("start_onboarding");
-      router.replace(newUrl.toString());
-
-      // Check if the user is logged in
-      if (user) {
-        // Start the onboarding process
-        setCurrentModal({ type: "onboarding" });
-      } else {
-        // Show the registration modal
-        setCurrentModal({ type: "signup" });
-      }
-    }
-  }, [searchParams, router, user, setCurrentModal]);
+  const { previousPath, currentPath } = useNavigation();
+  const pathname = usePathname();
+
+  // We want to avoid situations where a user skips the tutorial
+  // on the homepage, navigates directly to the questions feed,
+  // and then sees the tutorial pop up again (or vice versa).
+  // To handle this, we simply check that the previous page
+  // wasn’t the home or questions page.
+  const previousPathHasTutorial =
+    previousPath &&
+    ["/", "/questions/"].includes(
+      new URL(previousPath, process.env.APP_URL).pathname
+    );
 
   useEffect(() => {
-    handleOnboarding();
-  }, [handleOnboarding]);
+    // Checks if the hook has already been refreshed.
+    // Sometimes, it takes a moment for useNavigation to update from the previous route's values,
+    // so we need to perform this check to ensure we have updated values of previousPath
+    const hookUpdated = currentPath === pathname;
```

Comment:
@ncarazon This one is kinda workaround, could you please tell me if it's ok?

Reply:
The workaround looks legit. However, I see another potential issue with the logic to skip the onboarding flow: the onboarding flow will still be triggered after skipping it if navigated from a route different from home or questions feed. For example: home -> tournaments -> questions. Is it expected? Ideally, we would utilize React Context, that would store a flag for skipping the onboarding. This could also simplify the code of `OnboardingCheck`, as we wouldn't need to update `previousPathHasTutorial` each time we want to add an onboarding trigger on a new page.

Reply:
Yeah, this is what @SylvainChevalier wanted: we should encourage users to complete onboarding if they have not, so I assume we do like to show onboarding again

Reply:
@ncarazon It turned out skip button should permanently hide this tutorial, so no need to have this advanced previous page check. Thus, we don't need to hardcode list of pathes where this is used, so I assume we can keep this component as is without root layout